scrapy_poet/_request_fingerprinter.py

```diff
@@ -0,0 +1,171 @@
+try:
+    from scrapy.utils.request import RequestFingerprinter  # NOQA
+except ImportError:
+    from typing import TYPE_CHECKING
+
+    if not TYPE_CHECKING:
+        ScrapyPoetRequestFingerprinter = None
+else:
+    import hashlib
+    import json
+    from functools import cached_property
+    from logging import getLogger
+    from typing import Callable, Dict, List, Optional, get_args, get_origin
+    from weakref import WeakKeyDictionary
+
+    from scrapy import Request
+    from scrapy.crawler import Crawler
+    from scrapy.settings.default_settings import REQUEST_FINGERPRINTER_CLASS
+    from scrapy.utils.misc import create_instance, load_object
+    from web_poet import (
+        HttpClient,
+        HttpRequest,
+        HttpRequestBody,
+        HttpRequestHeaders,
+        PageParams,
+        RequestUrl,
+        Stats,
+    )
+
+    from scrapy_poet import InjectionMiddleware
+    from scrapy_poet.injection import get_callback
+
+    logger = getLogger(__name__)
+
+    def _serialize_dep(cls):
+        try:
+            from typing import Annotated
+        except ImportError:
+            pass
+        else:
+            if get_origin(cls) is Annotated:
+                annotated, *annotations = get_args(cls)
+                return f"{_serialize_dep(annotated)}{repr(annotations)}"
+        return f"{cls.__module__}.{cls.__qualname__}"
```

Comment:
Should we use https://github.com/scrapinghub/web-poet/blob/acef0a6a6d6dc8598176fc401d5785d1c9c4bf6a/web_poet/utils.py#L25?